src/nats.rs

```diff
@@ -0,0 +1,26 @@
+use serde::{Deserialize, Serialize};
+
+#[derive(Clone, Debug, Deserialize, Serialize)]
+#[serde(deny_unknown_fields, rename_all = "snake_case", tag = "strategy")]
+pub(crate) enum NatsAuthConfig {
+    UserPassword { user: String, password: String },
+    Token { token: String },
+    CredentialsFile { path: String },
+    NKey { nkey: String, seed: String },
+}
+
+impl NatsAuthConfig {
+    pub(crate) fn to_nats_options(&self) -> async_nats::Options {
+        match self {
+            NatsAuthConfig::UserPassword { user, password } => {
+                async_nats::Options::with_user_pass(user, password)
+            }
+            NatsAuthConfig::CredentialsFile { path } => async_nats::Options::with_credentials(path),
+            NatsAuthConfig::NKey { nkey, seed } => {
+                let kp = nkeys::KeyPair::from_seed(seed).unwrap();
+                async_nats::Options::with_nkey(nkey, move |nonce| kp.sign(nonce).unwrap())
```

Comment:
Is this `unwrap` safe? That is, could a problem with the key file cause vector to crash? If so, make this function return a `Result` and propagate the error to the caller.

Reply:
This unwrap is very not safe. I just copy/pasta'd this from the async-nats crate docs. I haven't actually properly tested NKey auth yet (will be adding an integration test for that). It's the main reason this PR is still a draft xD